scripts/29-35.R

```diff
@@ -0,0 +1,137 @@
+# Assignment 1:  
+library(tweedie) 
+library(ggplot2)
+library(parallel)
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+
+# Assignment 2:  
+MTweedieTests <-  
+  function(N,M,sig){ 
+    sum(replicate(M,simTweedieTest(N)) < sig)/M 
+  } 
+
+
+# Assignment 3:  
+df <-  
+  expand.grid( 
+    N = c(10,100,1000,5000, 10000), 
+    M = 1000, 
+    share_reject = NA) 
+
+
+#Use cluster to detect cores
+cl <- makeCluster(detectCores())
+
+# Use mclapply to parallelize the loop, with multiple cores
+df$share_reject <- mclapply(1:nrow(df), function(i) {
+  MTweedieTests(N = df$N[i], M = df$M[i], sig = 0.05)
+}, mc.cores = length(cl))
```

Comment:
I am not familiar with the mclapply() function, but the results suggest it is correct